plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts

```diff
@@ -311,25 +312,56 @@ export class NunjucksWorkflowRunner implements WorkflowRunner {
       const tmpDirs = new Array<string>();
       const stepOutput: { [outputName: string]: JsonValue } = {};
 
-      await action.handler({
-        input,
-        secrets: task.secrets ?? {},
-        logger: taskLogger,
-        logStream: streamLogger,
-        workspacePath,
-        createTemporaryDirectory: async () => {
-          const tmpDir = await fs.mkdtemp(`${workspacePath}_step-${step.id}-`);
-          tmpDirs.push(tmpDir);
-          return tmpDir;
-        },
-        output(name: string, value: JsonValue) {
-          stepOutput[name] = value;
-        },
-        templateInfo: task.spec.templateInfo,
-        user: task.spec.user,
-        isDryRun: task.isDryRun,
-        signal: task.cancelSignal,
-      });
+      const iterations = new Array<JsonValue>();
+      if (step.each) {
+        const each = await this.render(step.each, context, renderTemplate);
+        iterations.push(...each);
+      } else {
+        iterations.push({});
+      }
+
+      let actionInput = input;
+      for (const iteration of iterations) {
+        if (step.each) {
+          taskLogger.info(`Running step each: ${iteration}`);
+          context.each = iteration;
```

Comment:
I think that we probably want to create a new `context` for each run rather than mutating it.

We also probably want to have `value` and `key` as per the original comment. So that we can go through any iterable. https://github.com/backstage/backstage/issues/8890#issuecomment-1012377281

`[{ test: 'ben' }, { test: 'github'}]` would yield `{ value: { test: 'ben' }, key: 0 }` in the context for instance.

And providing an object like this: `{ test: 'ben', other: 'thing' }` would provide something like `{ value: 'ben', key: 'test' }` in the context.

Does that make sense?